native/mod_manager/context.c

```diff
@@ -72,48 +72,49 @@ static mem_t *create_attach_mem_context(char *string, int *num, int type, apr_po
 }
 
 /**
- * Insert(alloc) and update a context record in the shared table
+ * Update a context record in the shared table
  * @param pointer to the shared table.
  * @param context context to store in the shared table.
  * @return APR_SUCCESS if all went well
  *
  */
-static apr_status_t insert_update(void *mem, void **data, int id, apr_pool_t *pool)
+static apr_status_t update(void *mem, void *data, apr_pool_t *pool)
 {
-    contextinfo_t *in = (contextinfo_t *)*data;
+    contextinfo_t *in = (contextinfo_t *)data;
     contextinfo_t *ou = (contextinfo_t *)mem;
     (void)pool;
 
     if (strcmp(in->context, ou->context) == 0 && in->vhost == ou->vhost && in->node == ou->node) {
         /* We don't update nbrequests it belongs to mod_proxy_cluster logic */
         ou->status = in->status;
-        ou->id = id;
+        ou->id = in->id;
         ou->updatetime = apr_time_sec(apr_time_now());
-        *data = ou;
-        return APR_SUCCESS;
+        return APR_EEXIST; /* it exists so we are done */
     }
-    return APR_NOTFOUND;
+    return APR_SUCCESS;
```

Comment:
I am not sure if I understand this. If it does exist, it's updated and it returns `APR_EEXIST`. If it does not exist, it returns `APR_SUCCESS`. Would not it be better if in case it's updated the function would return `APR_SUCCESS` and `APR_NOT_FOUND` or `APR_ENOENT` in the other case?

Reply:
No see https://github.com/apache/httpd/blob/trunk/modules/slotmem/mod_slotmem_shm.c#L330

Reply:
Ok, but then I would change the comment

> @return APR_SUCCESS if all went well

to 

> @return APR_EEXIST if the record was updated, APR_SUCCESS otherwise